borgbase_repo.py

```diff
@@ -0,0 +1,495 @@
+#!/usr/bin/python
+
+from pprint import pprint
+
+ANSIBLE_METADATA = {
+		'metadata_version': '1.1',
+		'status': ['preview'],
+		'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: borgbase_repo
+
+short_description: Module for managing repos in borgbase.
+
+version_added: "2.4"
+
+description:
+		- "Module for managing repos in borgbase."
+
+options:
+		email:
+				description:
+						- The email address associated with the borgbase account
+				required: true
+		password:
+				description:
+						- The password for the borgbase account
+				required: true
+		state:
+				description:
+						- 'present' to ensure the repo exists, 'absent' to ensure it doesn't.
+				default: present
+				choices: [ absent, present ]
+		alert_days:
+				description:
+						- Number of days to send alerts if no activity detected
+				required: false
+		append_only:
+				description:
+						- True if repo is append only
+				required: false
+		append_only_keys:
+				description:
+						- List of keys for append only access
+				required: false
+		borg_version:
+				description:
+						- Version of borg to run on this repo
+				required: false
+		full_access_keys:
+				description:
+						- List of keys for full access
+				required: true
+		name:
+				description:
+						- Repo name
+				required: true
+		quota:
+				description:
+						- Disk quota for this repo (MB)
+				required: false
+		quota_enabled:
+				description:
+						- Whether quota is enabled for this repo
+				required: false
+		region:
+				description:
+						- Repo region
+				choices: [ eu, us ]
+				required: false
+				default: eu
+author:
+		- Andy Hawkins (@adhawkinsgh)
+'''
+
+EXAMPLES = '''
+- name: Read key from file
+  slurp:
+    src: ~/.ssh/id_rsa.pub
+  register: ssh_key
+  check_mode: yes
+
+- name: Create key
+  borgbase_ssh:
+    state: present
+    email: "{{ borgbase_email }}"
+    password: "{{ borgbase_password }}"
+    name: "{{ whoami.stdout }}@{{ ansible_hostname }}"
+    key: "{{ ssh_key['content'] | b64decode }}"
+  register: borgbase_key
+
+- name: Create repo
+  borgbase_repo:
+    state: present
+    email: "{{ borgbase_email }}"
+    password: "{{ borgbase_password }}"
+    name: "{{ ansible_hostname }}"
+    full_access_keys: [ "{{ borgbase_key.key_id }}" ]
+    quota_enabled: false
+    alert_days: 1
+  register: borgbase_repo
+
+- name: Set borgbase repo id
+  set_fact:
+    borgbackup_borgbase_repo: "{{ borgbase_repo.repo_id }}"
+
+- name: Set borgbackup_ssh_host
+  set_fact:
+    borgbackup_ssh_host: "{{ borgbackup_borgbase_repo }}.repo.borgbase.com"
+'''
+
+RETURN = '''
+repo_id:
+		description: The ID of the repo that was created or deleted
+		type: int
+		returned: always
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+REQUESTS_IMP_ERR = None
+try:
+	import requests
```

Comment:
`from ansible.module_utils.urls import open_url` could be used instead of `requests`.

Reply:
Thanks, I was unaware of this facility. I'll give it a try.

Reply:
Addressed in #7